docs/pages/components/checkbox.mdx

```diff
@@ -158,6 +166,66 @@ function() {
 }
 ```
 
+## Nested checkboxes
+
+Use 'indeterminate' prop to manage the behaviour of a parent checkbox in response to children checkboxes. This can help you create a nested selection menu.
+
+```jsx
+function() {
+  const [indeterminate, setIndeterminate] = React.useState(true)
+  const [parentChecked, setParentChecked] = React.useState(false)
+  const [checkedOne, setCheckedOne] = React.useState(true)
+  const [checkedTwo, setCheckedTwo] = React.useState(false)
+
+  const handleClick = (name) => {
+    if(checkedTwo && checkedOne || !checkedTwo && !checkedOne){
+        setIndeterminate(true)
+        setParentChecked(false)
+        }
```

Comment:
Hey, this code could be simplified a little. Here are the suggested change: 
```jsx
function() {
  const [checkboxA, setCheckboxA] = React.useState(true)
  const [checkboxB, setCheckboxB] = React.useState(false)

  const parentChecked = checkboxA && checkboxB
  const indeterminate = checkboxA || checkboxB

  return (
    <>
      <Field label="Parent checkbox">
        <Checkbox checked={parentChecked} indeterminate={indeterminate} />
      </Field>
      <Field label="checkbox A" ml="md">
        <Checkbox checked={checkboxA} onChange={() => setCheckboxA(!checkboxA)} />
      </Field>
      <Field label="checkbox B" ml="md">
        <Checkbox checked={checkboxB} onChange={() => setCheckboxB(!checkboxB)} />
      </Field>
    </>
  )
}
```

Also, the format is not right on your file. I suggest you to copy your code in a .jsx file, and to format it in vscode before moving it to the .mdx file.

Feel free to contact me if you need more explanations :)